test/e2e/funcs/feature.go

```diff
@@ -497,6 +498,100 @@ func ManagedResourcesOfClaimHaveFieldValueWithin(d time.Duration, dir, file, pat
 	}
 }
 
+// ListedResourcesValidatedWithin fails a test if the supplied list of resources
+// does not have the supplied number of resources that pass the supplied
+// validation function within the supplied duration.
+func ListedResourcesValidatedWithin(d time.Duration, list k8s.ObjectList, min int, validate func(object k8s.Object) bool, listOptions ...resources.ListOption) features.Func {
+	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		if err := wait.For(conditions.New(c.Client().Resources()).ResourceListMatchN(list, min, validate, listOptions...), wait.WithTimeout(d)); err != nil {
+			y, _ := yaml.Marshal(list)
+			t.Errorf("resources didn't pass validation: %v:\n\n%s\n\n", err, y)
+			return ctx
+		}
+
+		t.Logf("%d resource(s) have desired conditions", min)
+		return ctx
+	}
+}
+
+// ListedResourcesDeletedWithin fails a test if the supplied list of resources
+// is not deleted within the supplied duration.
+func ListedResourcesDeletedWithin(d time.Duration, list k8s.ObjectList, listOptions ...resources.ListOption) features.Func {
+	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		if err := c.Client().Resources().List(context.TODO(), list, listOptions...); err != nil {
```

Comment:
```suggestion
		if err := c.Client().Resources().List(ctx, list, listOptions...); err != nil {
```

Can we plumb this context up? There's a few other places (e.g. in `ListResourcesModifiedWith`) where this is relevant, too.